src/pages/sign-transaction/components/principal-value.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+import { useCurrentNetwork } from '@common/hooks/use-current-network';
+import { Link } from '@components/link';
+
+interface PrincipalValueProps {
+  address: string;
+}
+
+export function PrincipalValue(props: PrincipalValueProps): JSX.Element {
+  const { address, ...rest } = props;
+  const { mode } = useCurrentNetwork();
+  const url = `https://explorer.stacks.co/address/${address}?chain=${mode}`;
+
+  return (
+    <Link
+      fontSize={2}
+      fontWeight={500}
+      lineHeight="1.6"
+      wordBreak="break-all"
+      onClick={() => window.open(url, '_blank')}
```

Comment:
There's a method used elsewhere that uses `window.open` and validates the url. Can that be refactored out the componet and used here too?